dev/com.ibm.ws.microprofile.graphql.1.0/src/com/ibm/ws/microprofile/graphql/component/AuthInterceptor.java

```diff
@@ -0,0 +1,123 @@
+/*******************************************************************************
+ * Copyright (c) 2019 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.graphql.component;
+
+import java.lang.reflect.Method;
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.Priority;
+import javax.annotation.security.DenyAll;
+import javax.annotation.security.PermitAll;
+import javax.annotation.security.RolesAllowed;
+import javax.enterprise.context.Dependent;
+import javax.interceptor.AroundInvoke;
+import javax.interceptor.Interceptor;
+import javax.interceptor.InvocationContext;
+import javax.servlet.http.HttpServletRequest;
+
+import org.eclipse.microprofile.graphql.GraphQLApi;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+@Dependent
+@GraphQLApi
+@Interceptor
+@Priority(Interceptor.Priority.PLATFORM_BEFORE)
+public class AuthInterceptor {
+
+    private final static TraceComponent tc = Tr.register(AuthInterceptor.class);
+
+    @AroundInvoke
+    public Object checkAuthorized(InvocationContext ctx) throws Exception {
+        Method m = ctx.getMethod();
+
+        if (isAuthorized(m)) {
+            return ctx.proceed();
+        }
+        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            Tr.debug(tc, "Unauthorized!");
+        }
+        // TODO: determine if this is the right thing to do here... maybe different exception?
+        throw new Exception("Unauthorized");
+    }
+
+    private static boolean isAuthorized(Method m) {
+        // First check annotations on methods
+        if (m.getAnnotation(DenyAll.class) != null) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "@DenyAll on method");
```

Comment:
do we want to print the method name? Or do we want to avoid putting information about the user's app in the logs?

Reply:
the method name will be in the trace automatically because the `Method m` is a parameter to the method (the entry/exit trace is generated automatically)- this trace message is mainly intended to tell us that we found the annotation on the method - as opposed to the class.